src/main/scala/io/laserdisc/zio/oci/objectstorage/Live.scala

```diff
@@ -0,0 +1,67 @@
+package io.laserdisc.zio.oci.objectstorage
+
+import com.oracle.bmc.model.BmcException
+import com.oracle.bmc.objectstorage.ObjectStorageAsyncClient
+import com.oracle.bmc.objectstorage.responses.{GetObjectResponse, ListBucketsResponse, ListObjectsResponse}
+import com.oracle.bmc.objectstorage.requests.{GetObjectRequest, ListBucketsRequest, ListObjectsRequest}
+import com.oracle.bmc.responses.AsyncHandler
+import zio._
+import zio.blocking.Blocking
+import zio.duration._
+import zio.stream.{Stream, ZStream}
+
+import java.util.concurrent.{Future => JFuture}
+import java.io.IOException
+
+final class Live(unsafeClient: ObjectStorageAsyncClient) extends ObjectStorage.Service {
+  override def listBuckets(compartmentId: String, namespace: String): IO[BmcException, ListBucketsResponse] =
+    listBuckets(ListBucketsRequest.builder().compartmentId(compartmentId).namespaceName(namespace).build())
+
+  private def listBuckets(request: ListBucketsRequest): IO[BmcException, ListBucketsResponse] =
+    execute[ListBucketsRequest, ListBucketsResponse] { c =>
+      c.listBuckets(request, _: AsyncHandler[ListBucketsRequest, ListBucketsResponse])
+    }
+
+  override def listObjects(namespace: String, bucket: String): IO[BmcException, ListObjectsResponse] =
+    execute[ListObjectsRequest, ListObjectsResponse] { c =>
+      c.listObjects(
+        ListObjectsRequest.builder().bucketName(bucket).namespaceName(namespace).build(),
+        _: AsyncHandler[ListObjectsRequest, ListObjectsResponse]
+      )
+    }
+
+  override def getObject(namespace: String, bucket: String, name: String): ZStream[Blocking, BmcException, Byte] =
+    ZStream
+      .fromEffect(execute[GetObjectRequest, GetObjectResponse] { c =>
+        c.getObject(
+          GetObjectRequest.builder().namespaceName(namespace).bucketName(bucket).objectName(name).build(),
+          _: AsyncHandler[GetObjectRequest, GetObjectResponse]
+        )
+      })
+      .flatMap(is => ZStream.fromInputStreamEffect(IO(is.getInputStream()).refineToOrDie[IOException]))
+      .refineOrDie { case e: BmcException =>
+        e
+      }
+
+  def execute[I, O](f: ObjectStorageAsyncClient => Function1[AsyncHandler[I, O], JFuture[O]]): IO[BmcException, O] =
```

Comment:
The reasoning behind this signature is the following:

> Async client implementation for ObjectStorage service. 
> There are two ways to use async client:
> - Use AsyncHandler: using AsyncHandler, if the response to the call is an InputStream, like getObject Api in object storage service, developers need to process the stream in AsyncHandler, and not anywhere else, because the stream will be closed right after the AsyncHandler is invoked. 
> - Use Java Future: using Java Future, developers need to close the stream after they are done with the Java Future.

https://docs.oracle.com/en-us/iaas/tools/java/1.32.1/com/oracle/bmc/objectstorage/ObjectStorageAsyncClient.html